sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridClient.java

```diff
@@ -0,0 +1,550 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+package com.azure.messaging.eventgrid;
+
+import com.azure.core.annotation.Generated;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.exception.ResourceModifiedException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.Response;
+import com.azure.core.models.CloudEvent;
+import com.azure.core.util.BinaryData;
+import com.azure.messaging.eventgrid.models.AcknowledgeOptions;
+import com.azure.messaging.eventgrid.models.AcknowledgeResult;
+import com.azure.messaging.eventgrid.models.PublishResult;
+import com.azure.messaging.eventgrid.models.ReceiveResult;
+import com.azure.messaging.eventgrid.models.RejectOptions;
+import com.azure.messaging.eventgrid.models.RejectResult;
+import com.azure.messaging.eventgrid.models.ReleaseOptions;
+import com.azure.messaging.eventgrid.models.ReleaseResult;
+import java.time.Duration;
+import java.util.List;
+
+/** Initializes a new instance of the synchronous EventGridClient type. */
+@ServiceClient(builder = EventGridClientBuilder.class)
+public final class EventGridClient {
+
+    @Generated private final EventGridAsyncClient client;
+
+    /**
+     * Initializes an instance of EventGridClient class.
+     *
+     * @param client the async client.
+     */
+    @Generated
+    EventGridClient(EventGridAsyncClient client) {
```

Comment:
We should generate with `enable-sync-stack` flag set to true to use sync stack instead of sync-over-async.